src/objective/xentropy_objective.hpp

```diff
@@ -0,0 +1,240 @@
+        hessians[i] = static_cast<score_t>(z * (1.0f - z));
+      }
+    } else {
+      // compute pointwise gradients and hessians with given weights
+      #pragma omp parallel for schedule(static)
+      for (data_size_t i = 0; i < num_data_; ++i) {
+        const double z = 1.0f / (1.0f + std::exp(-score[i]));
+        gradients[i] = static_cast<score_t>((z - label_[i]) * weights_[i]);
+        hessians[i] = static_cast<score_t>(z * (1.0f - z) * weights_[i]);
+      }
+    }
+  }
+
+  const char* GetName() const override {
+    return "xentropy";
+  }
+
+  // convert score to a probability
+  void ConvertOutput(const double* input, double* output) const override {
+    output[0] = 1.0f / (1.0f + std::exp(-input[0]));
+  }
+
+  std::string ToString() const override {
+    std::stringstream str_buf;
+    str_buf << GetName();
+    return str_buf.str();
+  }
+
+  bool BoostFromAverage() const override { return true; }
+
+private:
+  /*! \brief Number of data points */
+  data_size_t num_data_;
+  /*! \brief Pointer for label */
+  const float* label_;
+  /*! \brief Weights for data */
+  const float* weights_;
+};
+
+/*!
+* \brief Objective function for alternative parameterization of cross-entropy (see top of file for explanation)
+*/
+class CrossEntropyLambda: public ObjectiveFunction {
+public:
+  explicit CrossEntropyLambda(const ObjectiveConfig&) {
+    min_weight_ = max_weight_ = 0.0f;
+  }
+
+  explicit CrossEntropyLambda(const std::vector<std::string>&) {
+  }
+
+  ~CrossEntropyLambda() {}
+
+  void Init(const Metadata& metadata, data_size_t num_data) override {
+    num_data_ = num_data;
+    label_ = metadata.label();
+    weights_ = metadata.weights();
+    // ensure that labels are in interval [0, 1], interval ends included
+    for (data_size_t i = 0; i < num_data; ++i) {
+      if (label_[i] < 0.0f || label_[i] > 1.0f) {
+        Log::Fatal("[%s]: does not tolerate label [#%i] outside [0, 1]",  GetName(), i);
+      }
+    }
+    Log::Info("[%s:%s]: labels passed interval [0, 1] check (objective)",  GetName(), __func__);
+    if (weights_ != nullptr) {
+      min_weight_ = weights_[0];
+      max_weight_ = weights_[0];
+      // ensure that every weight is strictly positive; and determine ratio : largest/smallest weight
+      for (data_size_t i = 0; i < num_data; ++i) {
+        if (weights_[i] <= 0.0f) {
+          Log::Fatal("[%s]: does not tolerate non-positive weight [#%i]", GetName(), i);
+        }
+        if (weights_[i] < min_weight_) min_weight_ = weights_[i];
+        if (weights_[i] > max_weight_) max_weight_ = weights_[i];
+      }
+      // Issue an info statement about this ratio; and possibly warn if excessively large ?!
+      double weight_ratio = max_weight_ / min_weight_;
+      Log::Info("[%s:%s]: min, max weights = %f, %f; ratio = %f", 
+                GetName(), __func__,
+                min_weight_, max_weight_,
+                weight_ratio);

```

Comment:
the same duplicated code like in ```metric``` . Maybe these checks can be define in ```LightGBM/utils/Common.h```, and all metrics and objective can call them.